vaccine_feed_ingest/runners/tx/arcgis/normalize.py

```diff
@@ -0,0 +1,229 @@
+
+
+def _get_id(site: dict) -> str:
+    data_id = site["attributes"]["OBJECTID"]
+
+    # Could parse these from directory traversal, but do not for now to avoid
+    # accidental mutation.
+    site = "arcgis"
+    runner = "tx"
+
+    # Could parse these from the input file name, but do not for now to avoid
+    # accidental mutation.
+    arcgis = "3078b524189848569f62985d71f4584b"
+    layer = 0
+
+    return f"{runner}:{site}:{arcgis}_{layer}:{data_id}"
+
+
+def _get_fallback_street(address: str) -> Optional[str]:
+    """
+    Extract the street from the address by taking everything up to the first comma
+    """
+    if not address:
+        return None
+    match = re.search(r"(.*?),", address, flags=re.IGNORECASE)
+    return match.group(1) if match else address
+
+
+def _get_fallback_zip(address: str) -> Optional[str]:
+    """
+    Extract the ZIP from the address if the last 5 characters are digits
+    """
+    if not address:
+        return None
+    match = re.search(r"([0-9]{5})$", address, flags=re.IGNORECASE)
+    return match.group(1) if match else None
+
+
+def _get_fallback_city(address) -> Optional[str]:
+    """
+    Attempt to extract the city from the address. First use the word immediately preceding ", TX" if present. As a
+    secondary fallback, return the word immediately preceding a 5 digit zipcode
+    """
+    if not address:
+        return None
+    match = re.search(r"([\S]*),\s?TX", address, flags=re.IGNORECASE)
+    if match:
+        return match.group(1)
+    match = re.search(r"([\S]*)\s[0-9]{5}", address, flags=re.IGNORECASE)
+    return match.group(1) if match else None
+
+
+def _get_address(site_attributes: dict) -> Optional[schema.Address]:
+    address = site_attributes["ADDRESS"]
+    street = (
+        site_attributes["STREET"]
+        if site_attributes["STREET"]
+        else _get_fallback_street(address)
+    )
+    city = (
+        site_attributes["CITY"]
+        if site_attributes["CITY"]
+        else _get_fallback_city(address)
+    )
+    zipcode = (
+        site_attributes["ZIP"] if site_attributes["ZIP"] else _get_fallback_zip(address)
+    )
+    try:
+        return schema.Address(
+            street1=street,
+            street2=None,
+            city=city,
+            state="TX",
+            zip=zipcode,
+        )
+    except ValidationError as e:
+        logging.error(
+            f"Unable to parse address: "
+            f"address={site_attributes['ADDRESS']}, "
+            f"street={site_attributes['STREET']}, "
+            f"city={site_attributes['CITY']}, "
+            f"zip={site_attributes['ZIP']}, "
+            f"e={e}"
+        )
+        return None
+
+
+def _get_contacts(site_attributes: dict) -> Optional[List[schema.Contact]]:
+    contacts = []
+    public_phone = site_attributes["PublicPhone"]
+    if public_phone:
+        source_phone = re.sub("[^0-9]", "", public_phone)
+        if len(source_phone) == 11:
+            source_phone = source_phone[1:]
```

Comment:
Is this case to handle phone numbers like `+1 (223) 445-6789`? If so -thank you! Could you be sure to follow the same output format as below (with the parens/space/dash)?

Reply:
done: https://github.com/CAVaccineInventory/vaccine-feed-ingest/pull/134/commits/484ff2a5766214024e39b6707117e17211771c6a